### ISSUE TITLE
<NODE_ENV>.json config need deep merge options

### ISSUE THREAD

**Issue by foxhound87** [feathersjs/feathers-configuration#16]
Issue related to: https://github.com/feathersjs/feathers-authentication/issues/191.

the problem is that if you have a `default.json` with

```
{ "facebook": { "secret": "TESTSECRET", "scope": ["email"] } }
```

and then another environment configuration (`production.json`) with

```
{ "facebook": { "secret": "SECRET_FROM_ENV" } }
```

It blows away the `scope` instead of merging it

**Comment by daffl** [comment]
I released version 0.3.0 (since it is a 0.x release it needs to be update in your `package.json` manually) with the fix. Hope everything is working for you now 😄

**Comment by foxhound87** [comment]
All is working now, thank you for the support!
